ic-agent/src/agent/mod.rs

```diff
@@ -1868,6 +1841,90 @@ impl<'agent> IntoFuture for UpdateBuilder<'agent> {
     }
 }
 
+/// HTTP client middleware. Implemented automatically for `reqwest`-compatible by-ref `tower::Service`, such as `reqwest_middleware`.
+pub trait HttpService: Send + Sync {
```

Comment:
Is there some reason we don't use `async_trait` here to make it simpler? We can work with it, but just wondering...

Reply:
No reason, I suppose.